contracts/Ops.sol

```diff
@@ -1,292 +1,291 @@
+    ) private returns (bytes32 taskId) {
+        taskId = LibTaskId.getTaskId(
+            _taskCreator,
             _execAddress,
-            _execSelector,
-            _resolverAddress,
-            _resolverData,
+            _execData.memorySliceSelector(),
+            _moduleData,
             _feeToken
         );
 
-        uint128 nextExec = uint256(_startTime) > block.timestamp
-            ? _startTime
-            : uint128(block.timestamp);
-
-        timedTask[taskId] = Time({nextExec: nextExec, interval: _interval});
-        emit TimerSet(taskId, nextExec, _interval);
-    }
+        require(
+            taskCreator[taskId] == address(0),
+            "Ops.createTask: Duplicate task"
+        );
 
-    /// @notice Create a task that tells Gelato to monitor and execute transactions on specific contracts
-    /// @dev Requires funds to be added in Task Treasury, assumes treasury sends fee to Gelato via Ops
-    /// @param _execAddress On which contract should Gelato execute the transactions
-    /// @param _execSelector Which function Gelato should eecute on the _execAddress
-    /// @param _resolverAddress On which contract should Gelato check when to execute the tx
-    /// @param _resolverData Which data should be used to check on the Resolver when to execute the tx
-    function createTask(
-        address _execAddress,
-        bytes4 _execSelector,
-        address _resolverAddress,
-        bytes calldata _resolverData
-    ) public override returns (bytes32 taskId) {
-        taskId = _createTask(
-            msg.sender,
+        LibTaskModule.onCreateTask(
+            taskId,
+            _taskCreator,
             _execAddress,
-            _execSelector,
-            _resolverAddress,
-            _resolverData,
-            address(0)
+            _execData,
+            _moduleData,
+            taskModuleAddresses
         );
-    }
 
-    /// @notice Create a task that tells Gelato to monitor and execute transactions on specific contracts
-    /// @dev Requires no funds to be added in Task Treasury, assumes tasks sends fee to Gelato directly
-    /// @param _execAddress On which contract should Gelato execute the transactions
-    /// @param _execSelector Which function Gelato should eecute on the _execAddress
-    /// @param _resolverAddress On which contract should Gelato check when to execute the tx
-    /// @param _resolverData Which data should be used to check on the Resolver when to execute the tx
-    /// @param _feeToken Which token to use as fee payment
-    function createTaskNoPrepayment(
-        address _execAddress,
-        bytes4 _execSelector,
-        address _resolverAddress,
-        bytes calldata _resolverData,
-        address _feeToken
-    ) public override returns (bytes32 taskId) {
-        taskId = _createTask(
-            msg.sender,
+        _createdTasks[_taskCreator].add(taskId);
+        taskCreator[taskId] = _taskCreator;
+        execAddresses[taskId] = _execAddress;
```

Comment:
We dont actually do anything with this state variable anymore, right? Or are we using it somewhere on the UI? If not, lets discuss whether we can cut this to reduce task creation overhead which is especially significant for use cases that use one-time tasks.

Reply:
These states are not used